subsys/net/lib/sockets/sockets_can.c

```diff
@@ -399,3 +394,14 @@ static const struct socket_op_vtable can_sock_fd_op_vtable = {
 	.getsockopt = can_sock_getsockopt_vmeth,
 	.setsockopt = can_sock_setsockopt_vmeth,
 };
+
+static bool check_socket_call(int family, int type, int proto)
```

Comment:
can_<supports/check/...>

Reply:
No need to introduce whole bloated function with prolog/epilog, the simple "if" check above can stay.